main.cpp

```diff
@@ -0,0 +1,59 @@
+#include <iostream>
+#include <random>
+#include <regex>
+
+void roll(int numberDice, int diceSides);
+void parseString(char *argument);
+
+int main(int argc, char *argv[]){
+//TODO Add some basic error checking
+    if(argc <= 1){
+        std::cout << "You haven't entered any arguments" << std::endl;
+        return -1;
+    }
+
+    for(int index = 1; index < argc; index++)
+         parseString(argv[index]);
+    return 0;
+}
+
+//Parses input then calls roll() on the arguments
+void parseString(char *argument){
+    //TODO Need to refactor, so unit testing is a viable strategy, possibly into class?
+    std::cmatch searchResults;
+
+    //Match any numbers at the begging of the line
+    std::regex firstNumberSearchString("^\\d+");
+
+    //Match any numbers at the end of line
+    std::regex secondNumberSearchString("\\d+$");
+
+    //Perform first search
+    regex_search(argument, searchResults, firstNumberSearchString);
+    int firstNumber = stoi(searchResults[0]);
```

Comment:
This will throw a std::out_of_range exception if the result is too large to fit in an integer (and that technically is possible given that you're looking for any number of digits with your regex). If it does throw an exception, the program will crash unless you add a try/catch block to handle it.